cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/CassandraCradleStorage.java

```diff
@@ -649,6 +689,30 @@ public TestEventsMessagesLinker getTestEventsMessagesLinker()
 						queryInfo));
 	}
 
+	@Override
+	protected CompletableFuture<Iterable<StoredTestEventMetadata>> doGetTestEventsMetadataAsync(
+			StoredTestEventId parentId, Instant from, Instant to) throws CradleStorageException
+	{
+		LocalDateTime fromDateTime = LocalDateTime.ofInstant(from, TIMEZONE_OFFSET),
+				toDateTime = LocalDateTime.ofInstant(to, TIMEZONE_OFFSET);
+		checkTimeBoundaries(fromDateTime, toDateTime, from, to);
+
+		LocalTime fromTime = fromDateTime.toLocalTime(),
+				toTime = toDateTime.toLocalTime();

```

Comment:
This part is duplicated multiple times. It is better to wrap this into an object, I think.